core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/detail/ReportDetailViewModel.kt

```diff
@@ -50,48 +63,85 @@ class ReportDetailViewModel(
                     text = systemImpl.getString(MR.strings.edit),
                     icon = FabUiState.FabIcon.EDIT,
                     onClick = this@ReportDetailViewModel::onClickEdit
-                )
+                ),
+                title = "Graph title",
             )
         }
-
-        launchIfHasPermission(
-            setLoadingState = true,
-            permissionCheck = { true }
-        ) {
-            launch {
-                navResultReturner.filteredResultFlowForKey(RESULT_KEY_REPORT_DETAIL)
-                    .collect { result ->
-                        val report = result.result as? Report ?: return@collect
-                        getReport(report)
+        viewModelScope.launch {
+            _uiState.whenSubscribed {
+                launchIfHasPermission(
+                    setLoadingState = true,
+                    permissionCheck = { true }
+                ) {
+                    val siteFlow =activeRepo.reportDao().findByUidLive(reportUid)
+                    launch {
+                        siteFlow.collect{
+                            val reportOptions = it?.let { optn ->
+                                optn.reportOptions?.takeIf { options -> options.isNotBlank() }
+                                    ?.let { options ->
+                                        Json.decodeFromString(
```

Comment:
Please use the json provided through the Dependency Injection (DI) - that is accessed using lower case ```json``` from a viewmodel